Replaced deprecated NSPasteboardNames with latest API constants

rr-codes opened the pull request:
#### c28bb7beb9af15b2b8f8ab858c71d892023cb7fa
<pre>
Replaced deprecated NSPasteboardNames with latest API constants
<a href="https://bugs.webkit.org/show_bug.cgi?id=241884">https://bugs.webkit.org/show_bug.cgi?id=241884</a>
rdar://95716307

Reviewed by Wenson Hsieh and Tim Horton.

NSGeneralPboard, NSFontPboard, NSRulerPboard, NSFindPboard and NSDragPboard were all deprecated in macOS 10.13.

* Source/WebCore/editing/cocoa/EditorCocoa.mm:
(WebCore::Editor::takeFindStringFromSelection):
(WebCore::Editor::replaceNodeFromPasteboard):
* Source/WebCore/editing/mac/EditorMac.mm:
(WebCore::Editor::pasteWithPasteboard):
* Source/WebCore/platform/mac/PasteboardMac.mm:
(WebCore::Pasteboard::createForCopyAndPaste):
(WebCore::Pasteboard::createForDragAndDrop):
* Source/WebKit/UIProcess/Cocoa/GlobalFindInPageState.mm:
(WebKit::findPasteboard):
* Source/WebKit/UIProcess/Cocoa/WebViewImpl.mm:
(WebKit::WebViewImpl::startDrag):
* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm:
(-[WKPDFLayerControllerDelegate writeItemsToPasteboard:withTypes:]):
(WebKit::PDFPlugin::handleEditingCommand):
* Source/WebKit/WebProcess/WebCoreSupport/mac/WebDragClientMac.mm:
(WebKit::WebDragClient::declareAndWriteDragImage):
* Source/WebKitLegacy/mac/Misc/WebNSPasteboardExtras.h:
* Source/WebKitLegacy/mac/Misc/WebNSPasteboardExtras.mm:
(+[NSPasteboard _web_setFindPasteboardString:withOwner:]):
(-[NSPasteboard _web_declareAndWriteDragImageForElement:URL:title:archive:source:]):
* Source/WebKitLegacy/mac/WebView/WebHTMLView.mm:
(-[WebHTMLView _fontAttributesFromFontPasteboard]):
(-[WebHTMLView copyFont:]):

Canonical link: <a href="https://commits.webkit.org/251834@main">https://commits.webkit.org/251834@main</a>
</pre>

hortont424 commented on Source/WebKitLegacy/mac/Misc/WebNSPasteboardExtras.h:
```diff
@@ -58,7 +58,7 @@ extern NSString *WebURLNamePboardType;
 - (void)_web_writeURL:(NSURL *)URL andTitle:(NSString *)title types:(NSArray *)types;
 
-// Sets the text on the NSFindPboard. Returns the new changeCount for the NSFindPboard.
+// Sets the text on the Find Pasteboard (NSPasteboardNameFind). Returns the new changeCount for the Find Pasteboard.
```
I would probably drop the caps on "find pasteboard", maybe?

pxlcoder commented on Source/WebCore/editing/cocoa/EditorCocoa.mm:
```diff
@@ -289,8 +289,8 @@
     types.append(String(legacyStringPasteboardType()));
     auto context = PagePasteboardContext::create(m_document.pageID());
     ALLOW_DEPRECATED_DECLARATIONS_BEGIN
```
Can remove the `ALLOW_DEPRECATED_DECLARATIONS_BEGIN` and `ALLOW_DEPRECATED_DECLARATIONS_END` now that we're no longer using the deprecated constants, here and everywhere else changes were made.

rr-codes committed:
Replaced deprecated NSPasteboardNames with latest API constants
https://bugs.webkit.org/show_bug.cgi?id=241884
rdar://95716307

Reviewed by Wenson Hsieh and Tim Horton.

NSGeneralPboard, NSFontPboard, NSRulerPboard, NSFindPboard and NSDragPboard were all deprecated in macOS 10.13.

* Source/WebCore/editing/cocoa/EditorCocoa.mm:
(WebCore::Editor::takeFindStringFromSelection):
(WebCore::Editor::replaceNodeFromPasteboard):
* Source/WebCore/editing/mac/EditorMac.mm:
(WebCore::Editor::pasteWithPasteboard):
* Source/WebCore/platform/mac/PasteboardMac.mm:
(WebCore::Pasteboard::createForCopyAndPaste):
(WebCore::Pasteboard::createForDragAndDrop):
* Source/WebKit/UIProcess/Cocoa/GlobalFindInPageState.mm:
(WebKit::findPasteboard):
* Source/WebKit/UIProcess/Cocoa/WebViewImpl.mm:
(WebKit::WebViewImpl::startDrag):
* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm:
(-[WKPDFLayerControllerDelegate writeItemsToPasteboard:withTypes:]):
(WebKit::PDFPlugin::handleEditingCommand):
* Source/WebKit/WebProcess/WebCoreSupport/mac/WebDragClientMac.mm:
(WebKit::WebDragClient::declareAndWriteDragImage):
* Source/WebKitLegacy/mac/Misc/WebNSPasteboardExtras.h:
* Source/WebKitLegacy/mac/Misc/WebNSPasteboardExtras.mm:
(+[NSPasteboard _web_setFindPasteboardString:withOwner:]):
(-[NSPasteboard _web_declareAndWriteDragImageForElement:URL:title:archive:source:]):
* Source/WebKitLegacy/mac/WebView/WebHTMLView.mm:
(-[WebHTMLView _fontAttributesFromFontPasteboard]):
(-[WebHTMLView copyFont:]):

Canonical link: https://commits.webkit.org/251834@main
